pydis_site/apps/api/models/bot/user.py

```diff
@@ -1,10 +1,22 @@
+from django.contrib.postgres.fields import ArrayField
+from django.core.exceptions import ValidationError
 from django.core.validators import MaxValueValidator, MinValueValidator
 from django.db import models
 
 from pydis_site.apps.api.models.bot.role import Role
 from pydis_site.apps.api.models.utils import ModelReprMixin
 
 
+def _validate_existing_role(value: int) -> None:
+    """Validate that a role exists when given in to the user model."""
+    role = Role.objects.filter(id=value)
+
+    if not role:
+        raise ValidationError(
+            f"Role with ID {value} does not exist"
+        )
```

Comment:
You're a bit generous with the line endings here (it fits on a single line that will only reach column 70), but I don't particularly mind.